drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -431,18 +467,23 @@ static void tx_timeout(struct k_timer *timer)
 		return;
 	}
 
-	struct uart_event evt = {
-		.type = UART_TX_ABORTED,
-		.data = {
-			.tx = {
-				.buf = data->tx_buf,
-				.len = 0
+	tx_complete(data);
+
+	if (txbuf == (void *)&data->txbyte) {
+		data->txbyte = -1;
+	} else {
+		struct uart_event evt = {
+			.type = UART_TX_ABORTED,
+			.data = {
+				.tx = {
+					.buf = data->tx_buf,
```

Comment:
Use `tx_buf` here, `data->tx_buf` has been changed in `tx_complete()`.

Reply:
good catch. fixed